dart-lang/webdev
# Run dwds tests in a copy of the sdk directory

**Test-only changes**

Run tests using a copy of the Dart SDK directory so we don't pollute the default one with generated assets. 

- Pass a `TestSdkConfigurationProvider` to`TestContext` contsructor.
- Make tests create and dispose of the SDK configuration provider as needed.
- Make sure all commands run from tests use dart executable from the SDK copy.

Closes: https://github.com/dart-lang/webdev/issues/1985
Towards: https://github.com/dart-lang/webdev/issues/1878
